book/src/guides/indent.md

```diff
@@ -46,6 +46,19 @@ capture on the same line, the indent level isn't changed at all.
 - `@outdent` (default scope `all`):
 Decrease the indent level by 1. The same rules as for `@indent` apply.
 
+- `@extend-indented`:
+Extend the range of this node to the end of the line and to lines that
+are indented more than the line that this node starts on. This is useful
```

Comment:
I'm curious if we can fork tree-sitter-python and have the function_definition and other rules that end in `:` consume the newline. That way the cursor would belong to the indented node and I think we might not need the extend system. I'll try it out

Reply:
I think the behavior would need to be a bit more complicated: It doesn't just have to consume the newline but also any following indented lines. Additionally, this should not happen when the function ends with for example a `return` expression.

If all this was implemented in the grammar, it would make this PR redundant but I'm not sure if it's worth the effort of forking the tree-sitter grammars for all languages that are affected by this.